### PR TITLE
Added case-insensitive matches for FreeBasic.

### PR DESCRIPTION
**Pull request by XusinboyBekchanov**
## Description

Added case-insensitive matches for FreeBasic heuristics.

## Checklist:

- [x] **I am fixing a misclassified language (FreeBasic)**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [x] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [ ] **I am changing the source of a syntax highlighting grammar**
  - Old: [URL to grammar repo]
  - New: [URL to grammar repo]

- [ ] **I am updating a grammar submodule**
  

- [ ] **I am adding new or changing current functionality**
  
  - [ ] I have added or updated the tests for the new or changed functionality.

- [ ] **I am changing the color associated with a language**
  
  - [ ] I have obtained agreement from the wider language community on this color change.
    - [URL to public discussion]
    - [Optional: URL to official branding guidelines for the language]

### CODE REVIEW

**Comment by DecimalTurn** [comment]
After looking a little more into it, I don't think it's a good idea to make the heuristics case-insensitive for FreeBasic mainly for these 2 reasons:
- When I look at [FreeBasic code](https://github.com/search?q=path%3A*.bas+AND+%2F%5E%5B+%5Ct%5D*%23%28%3F%3Adefine%7Cendif%7Cendmacro%7Cifn%3Fdef%7Cif%7Cinclude%7Clang%7Cmacro%29%5Cs%2F+language%3AFreeBasic&type=code&l=FreeBasic) on GitHub, I see that the use of lowercase is prevalent and I actually found no use of uppercase in the first five pages of results.
- VBA and VB6 use `#If` and so the case-insensitive version of the heuristic would match that and cause false positive for FreeBasic.

**Comment by XusinboyBekchanov** [comment]
> Do you have any examples of FreeBasic repos or files missclassified because they were using a different casing convention than the norm?

For example, here is a capital letter:
https://github.com/bob-the-hamster/ohrrpgce/blob/f8cb11c86de7eb1c9c32ae60bb7bd40f131ab97a/miditest.bas

**Comment by DecimalTurn** [comment]
> For example, here is a capital letter:
> [bob-the-hamster/ohrrpgce@f8cb11c/miditest.bas](https://github.com/bob-the-hamster/ohrrpgce/blob/f8cb11c86de7eb1c9c32ae60bb7bd40f131ab97a/miditest.bas)

I see that this file still makes use of lowercase `#include`, so the lowercase heuristics would still work.

However, after looking further, I've found the following file here:
https://github.com/doranchak/azdecrypt/blob/a9dbde3adc451c1aaccb4b4339f81e1ab5dafc0d/FreeBASIC-1.09.0-winlibs-gcc-9.3.0/examples/GUI/GTK%2B/FB_Calc/FB_Calc.bas#L3

It does only include the use of `#INCLUDE` and is indentified by Linguist as v7.25 as `VBA`, so I guess this would be a good case to justify the use of case-insensitive matches.

In that case, I'm in favor of this change.

File: lib/linguist/heuristics.yml [hunk 1]
```diff
@@ -89,7 +89,7 @@ disambiguations:
 - extensions: ['.bas']
   rules:
   - language: FreeBasic
-    pattern: '^[ \t]*#(?:define|endif|endmacro|ifn?def|if|include|lang|macro)\s'
+    pattern: '^[ \t]*#(?i)(?:define|endif|endmacro|ifn?def|include|lang|macro)\s'
```
Comment (Alhadis) [inline lib/linguist/heuristics.yml:92]:
```suggestion
    pattern: '^[ \t]*#(?i)(?:define|endif|endmacro|ifn?def|include|lang|macro)(?:$|\s)'
```

**Review by Alhadis — CHANGES_REQUESTED** [review]
While we're here, we might as well harden these heuristics to deal with files that aren't terminated by a trailing newline (something that's more common on Windows). For example:

~~~ruby
regex = /^[ \t]*#(?i)(?:define|endif|endmacro|ifn?def|include|lang|macro)\s/
"Foo\nBar\n#endif\n" =~ regex # => 8
"Foo\nBar\n#endif"   =~ regex # => nil
~~~

**Review by Alhadis — APPROVED** [review]

**Review by lildude — APPROVED** [review]
LGTM. Thanks.

Note: this PR will not be merged until close to when the next release is made. See [here](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes) for more details.

**Comment by XusinboyBekchanov** [comment]
Ok

**Review by lildude — APPROVED** [review]

**Pull request merged**